src/main/java/com/example/user_management/converter/UserConverter.java

```diff
@@ -0,0 +1,29 @@
+package com.example.user_management.converter;
+
+import com.example.user_management.entity.User;
+import com.example.user_management.model.request.CreateUserRequestModel;
+import org.springframework.stereotype.Component;
+
+import static org.springframework.util.Assert.hasLength;
+import static org.springframework.util.Assert.notNull;
+
+@Component
+public class UserConverter {
+
+    private UserConverter() {
+    }
+
+    public User toUser(final CreateUserRequestModel createUserRequestModel) {
+        assertCreateUserRequestModelExists(createUserRequestModel);
+        User user = new User();
+        user.setUsername(createUserRequestModel.getUsername());
+        user.setPassword(createUserRequestModel.getPassword());
+        return user;
+    }
+
+    private void assertCreateUserRequestModelExists(final CreateUserRequestModel createUserRequestModel) {
```

Comment:
Why converter takes care of validation for the request model? can't we create something like member function `validate()` within CreateUserRequestModel so model itself will take care of it's own validation and offload converter from doing that. Another point imagine the model can be used in other place/part of software where the client/user of the model won't be aware or won't perform any validation, so not to repeat validation code in different components it is yet another point to encapsulate logic within model.

Reply:
Added validation on CreateUserRequestModel attributes as annotation attribute